src/domains/dom_runner.h

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (C) 2023 Renesas Electronics Corporation.
+ * Copyright (C) 2023 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef DOMAIN_CONF_H_
+#define DOMAIN_C0NF_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if !defined(CONFIG_NATIVE_APPLICATION)
+extern struct xen_domain_cfg domd_cfg;
+#endif
+
+int create_domains();
```

Comment:
extern?

Reply:
hm - this is definition. It shouldn't be extern here AFAIK

Reply:
Hm, as far as I know extern keyword is used when we are going to use function or variable outside of the scope is was declared. Like it was described in the following link:https://www.scaler.com/topics/c-extern/
It says linker to take a look inside function table and find an address, associated with this name.
In this particular case we include header file so extern keyword is not needed because we directly include header. This is the common practice (Please, take a look in zephyr-xenlib headers). We have to use it when we are expecting code, which doesn't include our header directly, to use this definition. As it was done to domd_cfg, which is indirectly used by xen_cmds.c file.

Reply:
extern is redundant for func prototypes in any case
but I'm care here about missing `void` for function which doesn't have parameters: `int create_domains(void);`